commands/src/main/java/org/wildfly/extras/creaper/commands/ra/AddResourceAdapter.java

```diff
@@ -0,0 +1,82 @@
+package org.wildfly.extras.creaper.commands.ra;
+
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+/**
+ * Adds new resource adapter
+ */
+public class AddResourceAdapter implements OnlineCommand {
+    private String id = null;
+    private String module = null;
+    private String transactions = null;
+
+    private AddResourceAdapter(Builder builder) {
+        this.id = builder.id;
+        this.module = builder.module;
+        switch (builder.transactions) {
+            case NONE:
+                transactions = TransactionType.NONE.getValue();
+                break;
+            case XA:
+                transactions = TransactionType.XA.getValue();
+                break;
+            default:
+                transactions = TransactionType.NONE.getValue();
+                break;
+        }
+    }
+
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        Operations ops = new Operations(ctx.client);
+        Address address = Address.subsystem("resource-adapters").and("resource-adapter", id);
+        ops.add(address, Values.of("transaction-support", transactions).and("module", module));
+
+
+    }
+
+
+    public static final class Builder {
+        private  String id = null;
+        private String module = null;
+        private TransactionType transactions = null;
+
+        /**
+         * @param resourceAdapterId     name of RA
+         * @param resourceAdapterModule module with RA
+         * @param transactionSupport    type of transaction
+         */
+        public Builder(String resourceAdapterId, String resourceAdapterModule, TransactionType transactionSupport) {
+            if (resourceAdapterId == null) {
+                throw new IllegalArgumentException("resourceAdapterId must be specified");
+            }
+            if (resourceAdapterModule == null) {
+                throw new IllegalArgumentException("resourceAdapterModule must be specified");
+            }
+            if (transactionSupport == null) {
+                throw new IllegalArgumentException("transactionSupport must be specified");
+            }
+
+
+            this.id = resourceAdapterId;
+            this.module = resourceAdapterModule;
+            this.transactions = transactionSupport;
+        }
+
+        public AddResourceAdapter build() {
+            return new AddResourceAdapter(this);
+        }
+
+    }
+
+
+    @Override
+    public String toString() {
+        return "Add new Resource adapter: " + id;
```

Comment:
I like that you're providing `toString`, most people don't :-), but the common pattern is "CommandClassName $someIdentifierIfExists". In this case, I'd expect `return "AddResourceAdapter " + id;`